packages/babel-preset-env/src/index.js

```diff
@@ -230,12 +230,24 @@ export default declare((api, opts) => {
   }
 
   const plugins = [];
+  const overrides = [];
   const pluginUseBuiltIns = useBuiltIns !== false;
 
-  // NOTE: not giving spec here yet to avoid compatibility issues when
-  // transform-modules-commonjs gets its spec mode
   if (modules !== false && moduleTransformations[modules]) {
-    plugins.push([getPlugin(moduleTransformations[modules]), { loose }]);
+    // NOTE: not giving spec here yet to avoid compatibility issues when
+    // transform-modules-commonjs gets its spec mode
+    const item = [getPlugin(moduleTransformations[modules]), { loose }];
+
+    if (modules === "auto") {
+      // When the `exclude` callback returns a truthy value, Babel not use the
+      // 'item' plugin here.
+      overrides.push({
+        exclude: (filename, { caller }) => caller && caller.supportsStaticESM,
```

Comment:
Since the `caller` isn't file-specific but it is more similar to `env`, can we pass it using the `api` object instead using the `exclude` option, which is usally used to exclude plugins based on the file path?

Reply:
That's a great question. Let me revisit that. This approach was from an old branch I had, so I hadn't acutally revisited that. My concern would be caching of plugin instances. The issue with caching is that the API would need to be

```
const supportsStaticESM = api.caller(caller => !!(caller && caller.supportsStaticESM));
```
because your build might be used both with and without static ESM support, and Babel needs to know whether or not it should re-evaluate the plugin function, since it does its best to only evaluate the plugin/preset functions a single time unless told otherwise.

I'm actually fine if we want to expose the `api` approach, but I'd still probably be tempted to leave this exposed in `test/include/exclude` too.

Reply:
Caller is an object that can only contain `null`, `undefined`, booleans, strings, and numbers: it can be `JSON.stringify`ed and stored to invalidate the cache when needed.

Reply:
That's true, I'm personally not a huge fan of that because we could always want to expand that in the future, and having to re-stringify the object all the time just seems undesirable. At the end of the day I don't feel that strongly about it I guess. One other downside being that the plugin cache would get invalidated if _any_ caller flag changed, even if it was something other than `supportsStaticESM`.